server.go

```diff
@@ -1944,6 +1961,38 @@ func SendHeader(ctx context.Context, md metadata.MD) error {
 	return nil
 }
 
+// SetSendCompressor sets a compressor for outbound messages.
+// It may be called at most once, and must not be called after any event that
```

Comment:
> It may be called at most once

How do we ensure this? and what happens if it is called multiple times?

Reply:
My bad, we don't have any checks on it being called multiple times. The last valid value compressor is chosen if it is invoked multiple times. This must not be named after the header has been sent out. I have updated the doc.